dip-ct.md

```diff
@@ -0,0 +1,166 @@
+the amount being transferred.
+
+## Motivation
+
+Currently Dash transactions can optionally use CoinJoin to increase the privacy of transactions but it
+leaves much to be desired. This is because CoinJoin is a mixing protocol which
+is based on amount obfuscation which leaves addresses as public data.
+
+CoinJoin transactions leaks large amounts of transaction metadata which is
+accessible via public blockchain data and requires users to learn about various
+details and advanced options to use it in a privacy-preserving way. The current
+CoinJoin implementation also requires users to wait longer for
+increased privacy via more mixing rounds. Most users will have no idea how many rounds they should use or what the implications of this choice will be. This incentivizes users to use fewer mixing
+rounds to save time, reducing their privacy as well as the privacy of
+all users utilizing CoinJoins.
+
+## Conventions
+
+We will use these abbreviations:
+  * BP - Bulletproof, a type of size-optimized rangeproof
+  * BP+ - Bulletproof+
+  * BP++ - Bulletproof++
+  * CT - Confidential Transaction
+
+## Scope
+
+This DIP describes how CT can be implemented via the Dash Full Node,
+the implementation details of light wallet servers and clients and other nodes is out of scope.
+
+## Prior Work
+
+There are many different types of CTs. The term originally was used in the context
+of Bitcoin in 2013 and has grown to a research field with many flavors of CTs. Monero
+currently uses RingCT with Bulletproof+ optimizations and the purpose of this DIP is
+to decide exactly which kind of CT is appropriate for Dash. Monero first implemented
+RingCT in 2017 and then added Bulletproofs in 2018 which allow for an 80% reduction
+in size of transactions, reducing blockchain bloat as well as reducing transaction fees.
+A further improvement called Bulletproofs+ was completed in 2022 which further reduces
+transaction size by roughly 5-10% and improves speed by 10%. Yet another improvement
+called Bulletproofs++ is currently being worked on which further reduces transaction
+size and speeds up runtime. Since BP++ is still being actively worked on and has not
+yet been merged into Monero and BP+ are still not widely used it is currently recommended for Dash to use BPs.
+
+## Consensus Protocol Changes
+
+This DIP proposes a new transaction type as well as new address types and therefore is a consensus change.
+
+## Overview
+
+One of the large differences between Dash and Monero is the Elliptic Curve each is based on. Dash uses secp256k1
+which is inherited from Bitcoin Core while Monero uses the Curve25519 curve. Bulletproofs are curve-agnostic, they can be used with any elliptic curve, but when considering exactly how the code will be implemented and which low-level libraries to use, this becomes important. The Monero implementation of Bulletproofs is specific to the elliptic curve they use and can not easily be ported or used in the Dash codebase.
```

Comment:
> Bulletproofs are curve-agnostic

citation needed

> The Monero implementation of Bulletproofs is specific to the elliptic curve they use and can not easily be ported or used in the Dash codebase.

Not strictly relevant here

> which low-level libraries to use

A recommendation should be made re what library to use or could be used.

Reply:
@PastaPastaPasta these 3 comments are resolved in the latest version of the DIP